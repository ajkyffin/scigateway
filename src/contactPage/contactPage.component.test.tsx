import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import {
  ContactPageWithoutStyles,
  CombinedContactPageProps,
} from './contactPage.component';

describe('Contact page componet', () => {
  let shallow;
  let props: CombinedContactPageProps;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'ContactPage' });

    props = {
      res: undefined,
      classes: {
        root: 'root-class',
        container: 'container-class',
        titleText: 'titleText-class',
        contactDetails: 'contactDetails-class',
      },
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<ContactPageWithoutStyles {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
