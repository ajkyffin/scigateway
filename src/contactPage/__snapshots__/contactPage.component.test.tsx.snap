// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact page componet should render correctly 1`] = `
<div
  className="root-class"
>
  <WithStyles(ForwardRef(Typography))
    className="titleText-class"
    variant="h3"
  >
    title
  </WithStyles(ForwardRef(Typography))>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      contact-details-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="contactDetails-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "contact-details-description",
        }
      }
      variant="body1"
    />
  </div>
</div>
`;
